device_e2e/sync/test_sync_connect_disconnect.py

```diff
@@ -29,30 +31,35 @@ def test_sync_connect_disconnect(self, brand_new_client):
         client.connect()
         assert client.connected
 
+        leak_tracker.check_for_leaks()
+
     @pytest.mark.it("calls `on_connection_state_change` when setting the handler (connected case)")
     @pytest.mark.quicktest_suite
     def test_sync_on_connection_state_change_gets_called_with_current_state_connected(
-        self, brand_new_client
+        self, brand_new_client, leak_tracker
     ):
         """
         This test verifies that the `on_connection_state_change` handler gets called if
         the client is connected when the handler is set.
         """
+        leak_tracker.set_initial_object_list()
```

Comment:
Since this pattern recurs so much, could we potentially integrate it into the fixture, yield it, and then do the check at the end? That way it wouldn't have to be explicitly done for every test